drivers/dac/dac_demo/dac_demo.c

```diff
@@ -0,0 +1,124 @@
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <inttypes.h>
+#include <stdlib.h>
+#include <string.h>
+#include "dac_demo.h"
+#include "error.h"
+#include "util.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/***************************************************************************//**
+ * @brief init function for the dac demo driver
+ * @param desc - descriptor for the dac
+ * @param param - initialization param for dac
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t dac_demo_init(struct dac_demo_desc **desc,
+		      struct dac_demo_init_param *param)
+{
+	struct dac_demo_desc *adesc;
+	adesc = (struct dac_demo_desc*)calloc(1, sizeof(*adesc));
+
+	if(!adesc)
+		return -ENOMEM;
+
+	*desc = adesc;
+
+	return SUCCESS;
+}
+
+/***************************************************************************//**
+ * @brief free allocated resources
+ * @param desc - descriptor for the dac to be removed
+ * @return SUCCESS in case of success, FAILURE otherwise.
+*******************************************************************************/
+int32_t dac_demo_remove(struct dac_demo_desc *desc)
+{
+	if(!desc)
+		return -EINVAL;
+
+	free(desc);
+
+	return SUCCESS;
+}
+
+/***************************************************************************//**
+ * @brief read function for the dac demo driver
+ * @param desc - descriptor for the dac
+ * @param reg_index - the address at which we want to read
+ * @param readval- the value read from register
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t dac_demo_reg_read(struct dac_demo_desc *desc, uint8_t reg_index,
+			  uint8_t *readval)
+{
+	if(!desc || reg_index > ARRAY_SIZE(desc->reg))
+		return -EINVAL;
+
+	*readval = desc->reg[reg_index];
+
+	return SUCCESS;
+}
+
+/***************************************************************************//**
+ * @brief write function for the dac demo driver
+ * @param desc - descriptor for the dac
+ * @param reg_index - the address at which we want to write
+ * @param writeval - the value to be written
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t dac_demo_reg_write(struct dac_demo_desc *desc, uint8_t reg_index,
+			   uint8_t *writeval)
+{
+	if(!desc || reg_index > ARRAY_SIZE(desc->reg))
+		return -EINVAL;
+
+	desc->reg[reg_index] = *writeval;
+
+	return SUCCESS;
+}
```

Comment:
nitpick: no new line at end of file

Reply:
I think Antoniu wanted to say that you should add a new line at end of file to get rid of the GitHub warning.

Reply:
I misunderstood, it's corrected now